apps_product_creator/models/odoo_module.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2016-Today: Odoo Community Association (OCA)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import models, fields, api
+
+
+class OdooModule(models.Model):
+    _inherit = 'odoo.module'
+
+    product_template_id = fields.Many2one(
```

Comment:
The relation between _odoo.module_ and _product.template_ is a `'0..1 to 0..1'`, in the sense that one _odoo.module_ can have at most one _product.template_, and the same for the other way round. So here defining a  Many2one `product_template_id` is fine.

But  I'm thinking that converting the field `product_template_id` to a _One2many_ `product_template_ids` would result more clear and it also simplifies the code all around. For example in the `def _create_product()` you will not need to filter the matching product inside a loop, because you can read `product_template_ids` directly.

What do you think?

Reply:
I am not technically of the advantage but it looks confusing to me @astirpe